File: 251004/Ivanov/Lab3/discussion/src/main/java/com/example/rest/repository/PostRepository.java
package com.example.rest.repository;

import com.example.rest.entity.Post;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface PostRepository extends CassandraRepository<Post, String> {

    Optional<Post> findById(Long id);

    // Для поиска по country (если нужно)
    @Query("SELECT * FROM tbl_post WHERE country = ?0 ALLOW FILTERING")
    List<Post> findByCountry(String country);

    // Для поиска по content
    @Query("SELECT * FROM tbl_post WHERE content = ?0 ALLOW FILTERING")

    void deleteById(Long id);
}

File: 251001/Ostapchuk/discussion/src/main/kotlin/org/ex/discussion/repository/PostRepository.kt
package org.ex.discussion.repository

import org.ex.discussion.model.Post
import org.springframework.data.cassandra.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface PostRepository : CrudRepository<Post, Long> {

    @Query(
        value = "SELECT * FROM post WHERE id = :id ALLOW FILTERING;",
        allowFiltering = true
    )
    override fun findById(@Param("id") id: Long): Optional<Post>

    @Query(
        value = "DELETE FROM post WHERE id = :id AND country = :country;"
    )
    fun delete(@Param("id") id: Long, @Param("country") country: String)
}